source/lac/block_matrix_array.cc

```diff
@@ -426,4 +426,7 @@ template class BlockMatrixArray<double>;
 template class BlockTrianglePrecondition<float>;
 template class BlockTrianglePrecondition<double>;
 
+template class BlockMatrixArray<double, TrilinosWrappers::MPI::BlockVector>;
+template class BlockMatrixArray<float, TrilinosWrappers::MPI::BlockVector>;
```

Comment:
You'll need to guard this with something like `#ifdef DEAL_II_HAVE_TRILINOS` (compare the syntax in the other files in that directory) to ensure it also compiles for people without trilinos. You probably also want to instantiate it for the PETScWrappers block vectors.